src/Microsoft.Diagnostic.Utilities/Command/CommandProcessor.cs

```diff
@@ -0,0 +1,259 @@
+﻿// --------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//
+// --------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostic.Utilities
+{
+    public class CommandProcessor
+    {
+        private readonly IEnumerable<Assembly> _assemblies;
+        private Parser _parser;
+        private Command _rootCommand;
+
+        /// <summary>
+        /// Domain specific context passed to commands
+        /// </summary>
+        public object CommandContext { get; set; }
+
+        /// <summary>
+        /// Create an instance of the command processor;
+        /// </summary>
+        /// <param name="assemblies"></param>
+        public CommandProcessor(IEnumerable<Assembly> assemblies)
+        {
+            _assemblies = assemblies;
+        }
+
+        /// <summary>
+        /// Parse the command line.
+        /// </summary>
+        /// <param name="commandLine">command line txt</param>
+        /// <param name="console">option console</param>
+        /// <returns>exit code</returns>
+        public Task<int> Parse(string commandLine, IConsole console = null)
+        {
+            if (_parser == null)
+            {
+                var rootBuilder = new CommandLineBuilder();
+                rootBuilder.UseHelp()
```

Comment:
Did you see `UseDefaults`? It includes these and a few other setup calls on the builder.

Reply:
There were a few things I didn't think I wanted like "CancelOnProcessTermination" but it is probably because I didn't understand it completely. I was afraid it would affect my CTRL-C handling in the interactive keyboard input support. I'll try it out.

Reply:
`CancelOnProcessTermination` allows you to programmatically handle Ctrl-C (e.g. to do some cleanup before shutdown) but it doesn't change the behavior of your app.